src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -267,12 +267,19 @@ def api_calls(self, value: Optional[List[Dict[str, Any]]]) -> None:
     @property
     def snowflake_plan(self):
         """Convert to a SnowflakePlan"""
+        return self.get_snowflake_plan(skip_schema_query=False)
+
+    def get_snowflake_plan(self, skip_schema_query):
         if self._snowflake_plan is None:
             query = Query(self.sql_query, params=self.query_params)
             queries = [*self.pre_actions, query] if self.pre_actions else [query]
+            if skip_schema_query is True:
```

Comment:
```suggestion
            schema_query = self.schema_query if skip_schema_query else None
```
I feel None is better, the same below

Reply:
updated